<launch>

  <!-- Launch to training data from a directory using RoiPlayer -->
  <arg name="directory" default="kinect_labeled_data"/>
  <arg name="classifierDir" default="Kinect"/>
  <arg name="replayRate" default="5.0"/>
  <arg name="kinect_disparity_fix" default="true"/>

  <node name="Roi_Player" pkg="roiPlayer" type="roiPlayer">
        <param name="mode"            value="directory"/>
        <param name="imageFolderPath" value="$(env NIST_BAG_DIR)/$(arg directory)"/>
	<param name="replayRate"      value="$(arg replayRate)"/>	
  </node>

  <!-- Launch HaarDispAda Node-->
  <node name="HaarDispAdaNode" pkg="HaarDispAda" type="HaarDispAda174" output="screen">	
       <remap from="Color_Image" to="output_image"/>	
       <remap from="Disparity_Image" to="output_disparity"/>
       <remap from="input_rois" to="output_rois"/>
       <!-- mode={detect,train,load,accumulate} -->
       <param name="mode" value="accumulate"/>
       <param name="classifier_file" value="$(env NIST_CLASSIFIERS)/$(arg classifierDir)/HaarDispAda.xml"/>
       <param name="num_Training_Samples" value="13000"/>
       <param name="HaarDispAdaPrior" value="6.0"/>
       <param name="Kinect_Disparity_Fix" value="$(arg kinect_disparity_fix)"/>	
       <param name="UseMissingDataMask" value="false"/>
  </node> 

   <!-- display the rois after training-->
   <node pkg="roiViewer" type="roiViewer" name="roiViewer" >
        <param name="mode" type="string" value="roi_display"/> 
        <param name="label" type="int" value="1" />
	<remap from ="input_image" to="HaarDispAdaColorImage"/>
	<remap from ="input_rois" to="HaarDispAdaOutputRois"/>  
   </node>

</launch>